egs/wsj/s5/utils/prepare_extended_lang.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+# Copyright 2018  Xiaohui Zhang
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+# WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+# MERCHANTABLITY OR NON-INFRINGEMENT.
+# See the Apache 2 License for the specific language governing permissions and
+# limitations under the License.
+
+# This script adds word-position-dependent phones and constructs a host of other
+# derived files, that go in data/lang/.
+
+# Begin configuration section.
+oov_unigram_prob=0.0001 # Unigram LM probability we'll assign to each word outside the original vocab.
+srilm_opts="-subset -prune-lowprobs -unk"
+prep_lang_opts=
+lm=
+to_carpa=false # convert the language model to const arpa format rather than FST 
+stage=0
+# end configuration sections
+
+echo "$0 $@"  # Print the command line for logging
+
+. utils/parse_options.sh
+
+if [ $# -ne 7 ]; then
+  echo "usage: utils/prepare_extended_lang.sh <dict-src-dir> <oov-dict-entry> <extra-lexicon> <phone-symbol-table> <extended-dict-dir> <tmp-dir> <extended-lang-dir>"
+  echo "e.g.: utils/prepare_extended_lang.sh data/local/dict <SPOKEN_NOISE> lexicon_extra.txt data/lang/phones.txt data/local/dict_ext data/local/lang_ext data/lang_ext"
+  echo "The goal is to extend the lexicon from <dict-src-dir> with extra lexical entries from <extra-lexicon>, putting the extended lexicon into <extended-dict-dir>,"
+  echo "and then build a valid lang dir <extended-lang-dir>."
+  echo "<dict-src-dir> must be a valid dictionary dir and <oov-dict-entry> is the oov word (see utils/prepare_lang.sh for details)."
+  echo "If a language model is provided, we'll also extend the language model and re-build G.fst (or G.carpa), by inserting extra words as extra unigrams,"
```

Comment:
change "If a language model is provided," to "If a language model is provided via the --lm option,"